tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala

```diff
@@ -1103,6 +1103,26 @@ class ControlSyntaxSuite extends BaseDottySuite {
     )
   }
 
+  test("while-expr-do") {
+    val code =
+      """|  while (x > 0) && (y > 0) do
```

Comment:
should this also work:
```
  while (x > 0) && (y > 0) x += 1
```
also, is `do` allowed only in scala3?

Reply:
It shouldn't. It seems that the chained infix application works only if there is `do` keyword in the dotty parser.

Thanks for pointing at that. I will slightly rework that pr.

Reply:
Added checks on having `do`/`then` kw after complex expressions as it's works in scala3.